### ISSUE TITLE
Gotham Knights (1496790)

### ISSUE THREAD

**Issue by NTMan** [ValveSoftware/Proton#6256]
# Compatibility Report
- Name of the game with compatibility issues: Gotham Knights
- Steam AppID of the game: 1496790

## System Information
- GPU: AMD Radeon 6900XT
- Driver/LLVM version: [Mesa 22.3-git](https://gitlab.freedesktop.org/mesa/mesa/-/commit/1ae73bc0768cafa21897bced6d4d635059569fd1)/15.0.0
- Kernel version: 6.1-rc1
- Link to full system information report as [gist](https://gist.github.com/NTMan/33cca4906328793206218192820779ec)
- Proton version: 1666305600 experimental-bleeding-edge-7.0-27365-20221020-p082fae-wd1dbda-de311f2-v27072c

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1496790.log](https://github.com/ValveSoftware/Proton/files/9841300/steam-1496790.log)

## Symptoms 
After launch the game terminated with message "The application has hung and will now close. We apologize for the inconvenience."
![Screenshot from 2022-10-21 22-52-22](https://user-images.githubusercontent.com/200750/197258860-667e8300-b5bd-4d4c-8548-4a1dad287153.png)

## Reproduction
Always

**Comment by Newbytee** [comment]
Just curious, did you try the "Proton Hotfix" build that was mentioned by Plagman on Twitter?

**Comment by gofman** [comment]
Looks like GPU crash or hang:

> amdgpu: The CS has been rejected, see dmesg for more information (-14).

**Comment by NTMan** [comment]
> Looks like GPU crash or hang:
> 
> > amdgpu: The CS has been rejected, see dmesg for more information (-14).

```
[ 1370.725341] ------------[ cut here ]------------
[ 1370.725363] WARNING: CPU: 1 PID: 11960 at drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c:678 amdgpu_ttm_tt_get_user_pages+0x14c/0x190 [amdgpu]
[ 1370.725674] Modules linked in: uinput rfcomm snd_seq_dummy snd_hrtimer netconsole nft_objref nf_conntrack_netbios_ns nf_conntrack_broadcast nft_fib_inet nft_fib_ipv4 nft_fib_ipv6 nft_fib nft_reject_inet nf_reject_ipv4 nf_reject_ipv6 nft_reject nft_ct nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 ip_set nf_tables nfnetlink qrtr bnep sunrpc binfmt_misc intel_rapl_msr intel_rapl_common snd_seq_midi snd_seq_midi_event snd_hda_codec_realtek iwlmvm mt76x2u snd_hda_codec_generic snd_hda_codec_hdmi mt76x2_common edac_mce_amd snd_hda_intel mt76x02_usb snd_intel_dspcfg mt76_usb snd_intel_sdw_acpi mt76x02_lib snd_usb_audio snd_hda_codec uvcvideo mt76 kvm_amd snd_hda_core snd_usbmidi_lib videobuf2_vmalloc asus_ec_sensors mac80211 btusb videobuf2_memops snd_rawmidi snd_hwdep videobuf2_v4l2 kvm btrtl snd_seq btbcm snd_seq_device eeepc_wmi videobuf2_common iwlwifi libarc4 btintel asus_wmi irqbypass ledtrig_audio snd_pcm videodev xpad btmtk rapl joydev vfat sparse_keymap mc
[ 1370.725723]  platform_profile wmi_bmof ff_memless bluetooth snd_timer cfg80211 snd fat k10temp i2c_piix4 soundcore rfkill acpi_cpufreq zram hid_logitech_hidpp hid_logitech_dj amdgpu igb dca drm_ttm_helper ttm crct10dif_pclmul video crc32_pclmul crc32c_intel iommu_v2 polyval_clmulni gpu_sched polyval_generic drm_buddy ucsi_ccg nvme ghash_clmulni_intel typec_ucsi drm_display_helper sha512_ssse3 ccp typec nvme_core sp5100_tco cec nvme_common wmi ip6_tables ip_tables fuse
[ 1370.725762] CPU: 1 PID: 11960 Comm: GothamKnights-W Tainted: G        W    L    -------  ---  6.1.0-0.rc2.21.fc38.x86_64+debug #1
[ 1370.725767] Hardware name: System manufacturer System Product Name/ROG STRIX X570-I GAMING, BIOS 4403 04/27/2022
[ 1370.725772] RIP: 0010:amdgpu_ttm_tt_get_user_pages+0x14c/0x190 [amdgpu]
[ 1370.725958] Code: b9 ed e9 32 ff ff ff 4c 89 e9 89 ea 48 c7 c6 d0 c2 e1 c0 48 c7 c7 18 a0 02 c1 e8 cf 75 06 ee eb 8e 66 90 bd f2 ff ff ff eb 8d <0f> 0b eb f5 bd fd ff ff ff eb 82 bd f2 ff ff ff e9 62 ff ff ff 48
[ 1370.725963] RSP: 0018:ffff963516cffb00 EFLAGS: 00010282
[ 1370.725968] RAX: ffff88fc688ea528 RBX: ffff88fb82017400 RCX: 0000000000000008
[ 1370.725972] RDX: 0000000000000008 RSI: ffff88fc688ea528 RDI: ffff88ff9f9f3000
[ 1370.725976] RBP: 0000000000000001 R08: 0000000000000dc0 R09: 00000000ffffffff
[ 1370.725980] R10: ffff963516cffb38 R11: 0000000000000001 R12: ffff963516cffba8
[ 1370.725983] R13: 00000000593c0000 R14: ffff88fe0e8181e0 R15: ffff8901a283d000
[ 1370.725987] FS:  0000000054e0f6c0(0000) GS:ffff890a62e00000(0000) knlGS:0000000027dd0000
[ 1370.725991] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 1370.725996] CR2: 0000000028138030 CR3: 0000000740dfe000 CR4: 0000000000350ee0
[ 1370.726000] Call Trace:
[ 1370.726004]  <TASK>
[ 1370.726009]  amdgpu_cs_ioctl+0x87e/0x1fc0 [amdgpu]
[ 1370.726218]  ? amdgpu_cs_find_mapping+0xe0/0xe0 [amdgpu]
[ 1370.726396]  drm_ioctl_kernel+0xac/0x160
[ 1370.726405]  drm_ioctl+0x1e7/0x450
[ 1370.726410]  ? amdgpu_cs_find_mapping+0xe0/0xe0 [amdgpu]
[ 1370.726602]  amdgpu_drm_ioctl+0x4a/0x80 [amdgpu]
[ 1370.726777]  __x64_sys_ioctl+0x90/0xd0
[ 1370.726784]  do_syscall_64+0x5b/0x80
[ 1370.726791]  ? rcu_read_lock_sched_held+0x10/0x80
[ 1370.726797]  entry_SYSCALL_64_after_hwframe+0x63/0xcd
[ 1370.726802] RIP: 0033:0x7f8ba02cd04f
[ 1370.726807] Code: 00 48 89 44 24 18 31 c0 48 8d 44 24 60 c7 04 24 10 00 00 00 48 89 44 24 08 48 8d 44 24 20 48 89 44 24 10 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 18 48 8b 44 24 18 64 48 2b 04 25 28 00 00
[ 1370.726811] RSP: 002b:0000000054e0d4a0 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 1370.726833] RAX: ffffffffffffffda RBX: 0000000054e0d758 RCX: 00007f8ba02cd04f
[ 1370.726837] RDX: 0000000054e0d580 RSI: 00000000c0186444 RDI: 000000000000020b
[ 1370.726842] RBP: 0000000054e0d580 R08: 00007f87d004ae70 R09: 0000000054e0d540
[ 1370.726846] R10: 000000007fc03820 R11: 0000000000000246 R12: 00000000c0186444
[ 1370.726850] R13: 000000000000020b R14: 0000000054e0d730 R15: 0000000000000007
[ 1370.726865]  </TASK>
[ 1370.726868] irq event stamp: 0
[ 1370.726872] hardirqs last  enabled at (0): [<0000000000000000>] 0x0
[ 1370.726876] hardirqs last disabled at (0): [<ffffffffae0f6c48>] copy_process+0xa08/0x1ec0
[ 1370.726882] softirqs last  enabled at (0): [<ffffffffae0f6c48>] copy_process+0xa08/0x1ec0
[ 1370.726887] softirqs last disabled at (0): [<0000000000000000>] 0x0
[ 1370.726891] ---[ end trace 0000000000000000 ]---
[ 1370.726896] [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to process the buffer list -14!
```

Yet another game hit this annoying issue:
https://lore.kernel.org/lkml/9ab8a042-d826-c9b5-9942-26e7f415085d@leemhuis.info/T/ 

Full kernel log: 
[dmesg.txt](https://github.com/ValveSoftware/Proton/files/9855612/dmesg.txt)

**Comment by Mr-nUUb** [comment]
For me it refuses to play videos, like the intro video or some story related cut secenes. This does not mean that all cut scenes are affected, I think it's only affecting prerecorded videos, not scripted ingame stuff. Insted of the video, a generic "test image" (consisting of colors and noise in the lower right corner) is shown. The same "test image" is shown on top of the main menu:
![grafik](https://user-images.githubusercontent.com/36689675/198834788-417a15ca-87b4-43bc-a7b9-e26ad526a14b.png)
# System Information
- GPU: AMD Radeon 6900XT
- Driver/LLVM version: Mesa 22.1.7
- Kernel version: 6.0.2-2-MANJARO
- Link to full system information report as [gist](https://gist.github.com/Mr-nUUb/24898830d4e6620181d6356641c223b0)
- Proton version: 1666305600 experimental-bleeding-edge-7.0-27365-20221020-p082fae-wd1dbda-de311f2-v27072c

EDIT: crafting equipment also shows the same image (when craft is complete).
EDIT 2: added log file to gist. I think these lines are interesting:
```
winegstreamer error: decodebin0: Your GStreamer installation is missing a plug-in.
winegstreamer error: decodebin0: ../src-gst_base/gst/playback/gstdecodebin2.c(4719): gst_decode_bin_expose (): /GstBin:bin0/GstDecodeBin:decodebin0:
no suitable plugins found:
Couldn't set avdec_h264-0 to PAUSEDCouldn't set avdec_valve_h264-0 to PAUSED
```
EDIT 3: tried [proton-ge-custom-bin](https://aur.archlinux.org/packages/proton-ge-custom-bin) and videos work again, no gstreamer errors in logs. BTW: the video in the screenshot is the Gotham Knights logo.

**Comment by hakzsam** [comment]
@NTMan Are you still able to reproduce with up-to-date kernel and RADV?
@Mr-nUUb Is it OK for you now?

**Comment by Mr-nUUb** [comment]
> @Mr-nUUb Is it OK for you now?

It works for me because I switched to custom proton: [Proton GE](https://github.com/GloriousEggroll/proton-ge-custom) via [proton-ge-custom-bin](https://aur.archlinux.org/packages/proton-ge-custom-bin)

**Comment by tuxtergames** [comment]
Its broken after the last update, freeze on loading screen, even with proton GE, sometimes starting with this "VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json %command% DXVK_ASYNC=1 %command% WINE_FULLSCREEN_FSR=1 %command% DXVK_FRAME_RATE=144 %command% mangohud %command% -useallavailablecores -notexturestreaming" taking from protondb

**Comment by Mr-nUUb** [comment]
> Its broken after the last update, freeze on loading screen, even with proton GE, sometimes starting with this "VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json %command% DXVK_ASYNC=1 %command% WINE_FULLSCREEN_FSR=1 %command% DXVK_FRAME_RATE=144 %command% mangohud %command% -useallavailablecores -notexturestreaming" taking from protondb

Is this one big command or are these multiple commands? Because it looks like multiple commands but it's wrapped in one pair of double quotes. Allow me to fix this mess:
```
VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json DXVK_ASYNC=1 WINE_FULLSCREEN_FSR=1 DXVK_FRAME_RATE=144 mangohud %command% -useallavailablecores -notexturestreaming
```

With Proton 7.0-4, default and custom parameters, I cannot start the game.
With Proton Experimental and Proton GE 7-41, default and custom parameters, I can start the game and load into the Belfry just fine.

TL;DR: cannot reproduce.

PS: Proton Experimental plays prerecorded videos just fine, no test image anymore.

EDIT: specifying the Vulkan ICD shouldn't be necessary, except you have either multiple graphics cards from different manufacturers (AMD/Intel/Nvidia) or both RADV and AMDVLK installed.

**Comment by Mte90** [comment]
Last proton next that should fix issues well doesn't fix them.
I still have crashes, as example yesterday I was in a crash loop on a cutscene (the one about Harley and monarch theatre) that was just crashing after few seconds. On next start of the game again and so on, I was able to escape it just opening the game menu and the game didn't crashed and the cutscene ended with no issues.
The game crashes randomly, like after 2 hours or less.

All those cases on Steam Deck.

**Comment by toalex77** [comment]
Currently, I've collected 4 log files: 2 with Proton GE and 2 with Proton Experimental.  

In 3 of them the game crashed after some hours of game, but by this time I avoid behaviours that I know can cause crash, as for example the use of hte BatCycle, or moving too fast in the environment.

Every time the game crash there is a message in the log, like this:
`warn:seh:OutputDebugStringA "LowLevelFatalError [File:Unknown] [Line: 881] \nRendering thread exception:\r\nFatal error!\r\n\r\nUnhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffff\r\n\r\n\n"`

If it can be useful, I can upload all 4 log files.

I'm on openSUSE Tumbleweed, kernel 6.1.8, AMD Opteron CPU, AMD RX 6600 GPU with Mesa 22.3.4 drivers. If other informations are needed, I can upload them as well.

**Comment by Mte90** [comment]
To avoid some crashes on Windows/linux there is an unofficial patch to the game executable https://steamcommunity.com/app/1496790/discussions/0/3761101693152548197/
On my usage seems that removed the crashes on SteamDeck at all.

**Comment by toalex77** [comment]
> To avoid some crashes on Windows/linux there is an unofficial patch to the game executable https://steamcommunity.com/app/1496790/discussions/0/3761101693152548197/ On my usage seems that removed the crashes on SteamDeck at all.

I tried it, but with no success. The game has continued to crash (but I haven't logs with the patch applied), so after a while a removed it.

**Comment by toalex77** [comment]
> To avoid some crashes on Windows/linux there is an unofficial patch to the game executable https://steamcommunity.com/app/1496790/discussions/0/3761101693152548197/ On my usage seems that removed the crashes on SteamDeck at all.

@Mte90 Hi, I decided to give a second try to the unofficial patch. From the first time that I tried it, has passed some time, and in the meanwhile a new kernel, a new version of Mesa, and also some new version of Proton are released.

I've played several hours, I've also incresed my graphics details and used the BatCycle for 10-15 minutes without interruption (I know... it was really boring), and until now the game has not crashed anymore.

Maybe, this unofficial patch has solved also my issues. I hope.

I will update this issue if there is any further news (or crashes).

Thanks.

**Comment by Mte90** [comment]
Reading about the bug in this case was something Audio about not finding something, I read also about other kind of crashes but this game isn't getting any updates since months so I suggest just to apply that patch and see if others just will be published online by users.

I see sometimes now that when I think was crashing the mission is not marked rightly (in the map when you go to the markers, not the main/side mission, just the city one), so the mission is not marked as finished.

PS: batcycle is useless, I never used it and I am now in New game+.

**Comment by toalex77** [comment]
Yesterday, finally, they have updated the game with a bugfix release, and now the game as returned to stutter and crash as before. Sadly the unofficial patch cannot be applied anymore.

**Comment by Mte90** [comment]
Just for reference https://steamcommunity.com/app/1496790/eventcomments/3773490215216701885?snr=1_5_9_ they fixed some crashes but not all of them I guess.

**Comment by Wemmy0** [comment]
I can't even get the game to start opening. Just stuck on "Launching..."

**Comment by TheAquabat** [comment]
can someone confirm ? is this broken when they pulled an update with Epic Online Services ? it doesnt work with Proton Experimental either. Just saying because this game is marked as verified and playable on the deck and its not working for me on ArchLinux.
EpicOnlineServices.exe is either not found, a message appears, or stuck at 100% and the game doesnt launch.

**Comment by Mr-nUUb** [comment]
Broken on Arch Linux, complains about missing "Epic Online Services".
![grafik](https://github.com/ValveSoftware/Proton/assets/36689675/6052750f-4e42-4044-9e23-161efd1bb1c8)
Tested with Proton-GE, Proton Experimental and Proton 8.0-4.

**Comment by TheAquabat** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6256#issuecomment-1818013074

Ok I made it work, not sure how. But in my case I think it was that for some reason, EOS doesnt like ntfs or exfat partitions, not really sure... but now it worked with Proton experimental with an ext4 partition, and also I had to nuke the pfx prefix. Maybe that was the problem and with this new update the pfx prefix needs to be nuked. Maybe the issue with this game can be fixed with protontricks ?

**Comment by Mr-nUUb** [comment]
Today Steam installed an update for this game, 574,3 MB in size. I can start the game again, but it takes ages. Main menu and gameplay works. Exiting the game does not work, it just freezes.

EDIT: will upload logs later, its getting too late here.

**Comment by ipr** [comment]
There is a regression in this game that causes GPU crash/hang when starting (before seeing main menu). Sympoms are that screen goes black and GPU cooling fans are blowing at full speed while system is unresponsive. Reset via computer reset-button does not work and computer needs to be powered off to restore display.

I haven't played this game for a while (lack of disk space), but problem was there with Mesa 24.3 and still exists with Mesa 25.0. Problem appears with both Proton 9.0 and Proton experimental. Kernel version 6.13.0 upwards.

Gpu is: 0a:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Vega 10 XL/XT [Radeon RX Vega 56/64] (rev c3)
Mesa: 25.0.1-2 
Kernel: Linux version 6.13.6


Log shows following lines:
```
719.810:02a4:0460:warn:vkd3d-proton:vkd3d_native_sync_handle_release: Failed to release semaphore (#12a).
719.810:02a4:0460:warn:vkd3d-proton:dxgi_vk_swap_chain_wait_worker: Failed to increment swapchain semaphore. Did application forget to acquire?
```

Another interesting bit (maybe not):
```
info:  Detected XeSS usage, hiding Intel GPU Vendor
```
There is no Intel GPU in the system, just ethernet controller, so no idea why it even logs that.

[steam-1496790.log.gz](https://github.com/user-attachments/files/19142243/steam-1496790.log.gz)

**Comment by ipr** [comment]
Looking at the log there is earlier error:
```
MESA: error: DRM_IOCTL_VIRTGPU_GET_CAPS failed with Invalid argument
MESA: error: DRM_IOCTL_VIRTGPU_CONTEXT_INIT failed with Invalid argument, continuing without context...
MESA: error: DRM_VIRTGPU_RESOURCE_CREATE_BLOB failed with Invalid argument
MESA: error: Failed to create virtgpu AddressSpaceStream
MESA: error: vulkan: Failed to get host connection
```

**Comment by ipr** [comment]
Also, this is NOT running in a VM so no idea why it would even try to use virtio in this case.

**Comment by simifor** [comment]
@ipr the game is working fine for me on a selfcompiled build of mesa, but given that mesa 25 is still an early release after all, could you try downgrading it to something like 24.3.4 and see if the issue persists? Regarding it misdetecting an intel gpu, does your system use an intel cpu?

**Comment by ipr** [comment]
> [@ipr](https://github.com/ipr) the game is working fine for me on a selfcompiled build of mesa, but given that mesa 25 is still an early release after all, could you try downgrading it to something like 24.3.4 and see if the issue persists? Regarding it misdetecting an intel gpu, does your system use an intel cpu?

Like I said, problem already existed in 24.3. I only listed the most recent version I've tried with.

And like I said, only Intel chip is the ethernet controller.

Edit: there is another game using Unreal Engine that crashes same way (same symptoms) but is far less reproducible at random places. Both have had same problem with earlier Mesa and earlier kernels. In this case it happens consistently at same place every time.

**Comment by simifor** [comment]
My apologies, between reading your report and testing the game I got some stuff mixed. So you have two UE games with related issues, is the other one also a dx12 game? Have you played UE games that use dx11? or dx12 games that don't use UE?

The xess message you see in your log is unrelated to your issue. Your system going unresponsive seems more interesting, and that's something that should leave messages in your kernel logs. If you are on a systemd distro, you can use `journalctl -k` to get this information, you can print it to a file with the following `journalctl -k > kernel.log` though I would suggest filtering the logs by boot session with `-b`, assuming you play the game and then reboot the system, you'd use `journalctl -k -b -1 > kernel.log` to get the kernel logs from the previous session (-2 would be the session before the previous one and so on).

**Comment by ipr** [comment]
I assume the other one uses dx12, not certain about it. I haven't had much chance to look for different cases since I need to use the computer for other things as well.

There is one error in the logs:
```
kernel: snd_hda_intel 0000:0a:00.1: Unable to change power state from D3hot to D0, device inaccessible
kernel: snd_hda_intel 0000:0a:00.1: CORB reset timeout#2, CORBRP = 65535
```

That might indicate some kind of power management issue in the system? Note that there is no Intel-chip for the audio, but it might be compatible that lspci reports this:
```
0c:00.4 Audio device: Advanced Micro Devices, Inc. [AMD] Starship/Matisse HD Audio Controller
        Subsystem: Gigabyte Technology Co., Ltd Device a0cf
        Flags: bus master, fast devsel, latency 0, IRQ 104, IOMMU group 29
        Memory at fcd00000 (32-bit, non-prefetchable) [size=32K]
        Capabilities: <access denied>
        Kernel driver in use: snd_hda_intel
        Kernel modules: snd_hda_intel
```
Audio uses chip (and connection) on the motherboard, NOT HDMI on the display card.

So it might not be related to dxvk, or proton, but something is confusing the power management of the whole system? 

There is another audio device on the system. That is not supposed to be used, but audio driver is the same:
```
0a:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Vega 10 HDMI Audio [Radeon Vega 56/64]
        Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] Vega 10 HDMI Audio [Radeon Vega 56/64]
        Flags: bus master, fast devsel, latency 0, IRQ 103, IOMMU group 24
        Memory at fcea0000 (32-bit, non-prefetchable) [size=16K]
        Capabilities: <access denied>
        Kernel driver in use: snd_hda_intel
        Kernel modules: snd_hda_intel
```
The monitors don't have speakers in them so it would not make any sense to route audio through HDMI anyway..

**Comment by simifor** [comment]
@ipr can you attach the whole log? and if you name the game I could tell you if it uses dx12, or you can run it with mangohud which can tell you if you're using vkd3d (which is used for dx12)

**Comment by ipr** [comment]
> [@ipr](https://github.com/ipr) can you attach the whole log? and if you name the game I could tell you if it uses dx12, or you can run it with mangohud which can tell you if you're using vkd3d (which is used for dx12)

Full log is in comment above: https://github.com/ValveSoftware/Proton/issues/6256#issuecomment-2708094493

That said, the power state -related message lead me to check a few things and I found that I had an udev-rule that was setting "high" performance mode on the GPU. Changing it to "auto" did not work either so I deleted that entire rule and now Gotham Knights passes the part about setting initial configuration at beginning and into the main menu without crashing.

Basically, this was the notable hint that Vega-cards have something strange with power management: https://wiki.archlinux.org/title/AMDGPU#System_freeze_or_crash_when_gaming_on_Vega_cards

Looks like you can't change it from the bootup-state at all or it becomes unstable - you can't force it into another mode with power_dpm_force_performance_level
